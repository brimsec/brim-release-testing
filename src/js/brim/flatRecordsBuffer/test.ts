import brim from ".."

test("coverts to an array of records", () => {
  const buffer = brim.flatRecordsBuffer()
  buffer.add(0, records)
  expect(buffer.channels()[0].records()).toMatchSnapshot()
})

test("flattens the columns", () => {
  const buffer = brim.flatRecordsBuffer()
  buffer.add(0, nestedRecords)
  expect(buffer.columns()).toMatchSnapshot()
})

test("coverts to an array of nested records", () => {
  const buffer = brim.flatRecordsBuffer()
  buffer.add(0, nestedRecords)
  expect(buffer.channels()[0].records()).toMatchSnapshot()
})

test("triple nested", () => {
  const buffer = brim.flatRecordsBuffer()
  buffer.add(0, tripleNest)
  expect(buffer.channels()[0].records()).toEqual([
    [
      {
        name: "a.b.c",
        type: "addr",
        value: "192.168.0.1"
      }
    ]
  ])
})

const records = [
  {
    id: 0,
    type: [
      {name: "ts", type: "time"},
      {name: "_path", type: "string"},
      {name: "count", type: "count"}
    ],
    values: ["1428917670.000000000", "dns", "2"]
  },
  {id: 0, values: ["1428917670.000000000", "conn", "3"]},
  {id: 0, values: ["1428917670.000000000", "capture_loss", "1"]},
  {id: 0, values: ["1428917640.000000000", "conn", "1"]}
]

const nestedRecords = [
  {
    id: 10,
    type: [
      {name: "_path", type: "string"},
      {name: "ts", type: "time"},
      {name: "uid", type: "string"},
      {
        name: "id",
        type: [
          {name: "orig_h", type: "addr"},
          {name: "orig_p", type: "port"},
          {name: "resp_h", type: "addr"},
          {name: "resp_p", type: "port"}
        ]
      },
      {name: "proto", type: "enum"},
      {name: "service", type: "string"},
      {name: "duration", type: "interval"},
      {name: "orig_bytes", type: "count"},
      {name: "resp_bytes", type: "count"},
      {name: "conn_state", type: "string"},
      {name: "local_orig", type: "bool"},
      {name: "local_resp", type: "bool"},
      {name: "missed_bytes", type: "count"},
      {name: "history", type: "string"},
      {name: "orig_pkts", type: "count"},
      {name: "orig_ip_bytes", type: "count"},
      {name: "resp_pkts", type: "count"},
      {name: "resp_ip_bytes", type: "count"},
      {name: "tunnel_parents", type: "set[string]"}
    ],
    values: [
      "conn",
      "1428917684.732640",
      "ClM1vE3c4iNZUgpZhh",
      ["fe80::eef4:bbff:fe4f:b245", "143", "ff02::16", "0"],
      "icmp",
      null,
      null,
      null,
      null,
      "OTH",
      "F",
      "F",
      "0",
      null,
      "1",
      "76",
      "0",
      "0",
      null
    ]
  }
]

const tripleNest = [
  {
    id: 0,
    type: [{name: "a", type: [{name: "b", type: [{name: "c", type: "addr"}]}]}],
    values: [[["192.168.0.1"]]]
  }
]
