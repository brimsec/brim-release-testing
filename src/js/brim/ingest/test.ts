import ingest from "./"

test("one pcap default", () => {
  const data = ingest.getParams([{type: "pcap", path: "/work/my.pcap"}])

  expect(data).toEqual({
    dataDir: "",
    name: "my.pcap",
    endpoint: "pcap",
    paths: ["/work/my.pcap"]
  })
})

test("one zeek log default", () => {
  const data = ingest.getParams([{type: "log", path: "/work/zeek.log"}])

  expect(data).toEqual({
    name: "zeek.log",
    dataDir: "",
    endpoint: "log",
    paths: ["/work/zeek.log"]
  })
})

test("two zeek logs in same dir default", () => {
  const data = ingest.getParams([
    {type: "log", path: "/work/zeek-1.log"},
    {type: "log", path: "/work/zeek-2.log"}
  ])

  expect(data).toEqual({
    name: "work",
    dataDir: "",
    endpoint: "log",
    paths: ["/work/zeek-1.log", "/work/zeek-2.log"]
  })
})

test("two zeek logs in different dir default", () => {
  const data = ingest.getParams(
    [
      {type: "log", path: "/work/day-1/zeek.log"},
      {type: "log", path: "/work/day-2/zeek.log"}
    ],
    "",
    [],
    new Date(0)
  )

  expect(data).toEqual({
    name: "zeek_1969-12-31_16:00:00",
    dataDir: "",
    endpoint: "log",
    paths: ["/work/day-1/zeek.log", "/work/day-2/zeek.log"]
  })
})

test("two pcaps", () => {
  const data = ingest.getParams([
    {type: "pcap", path: "/pcap-1"},
    {type: "pcap", path: "/pcap-2"}
  ])

  expect(data).toEqual({
    error: "Only one pcap can be opened at a time."
  })
})

test("1 pcap and 1 zeek", () => {
  const data = ingest.getParams([
    {type: "pcap", path: "/pcap-1"},
    {type: "log", path: "/zeek-1"}
  ])

  expect(data).toEqual({
    error: "Only one pcap can be opened at a time."
  })
})
