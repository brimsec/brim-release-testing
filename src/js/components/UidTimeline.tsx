import {connect} from "react-redux"
import React from "react"
import classNames from "classnames"
import * as d3 from "d3"
import isEqual from "lodash/isEqual"

import {DispatchProps} from "../state/types"
import {UID_CORRELATION_LIMIT} from "../searches/programs"
import {submitSearch} from "../flows/submitSearch/mod"
import {viewLogDetail} from "../flows/viewLogDetail"
import Log from "../models/Log"
import SearchBar from "../state/SearchBar"
import brim from "../brim"
import dispatchToProps from "../lib/dispatchToProps"

type OwnProps = {
  logs: Log[]
  log: Log
}

type Props = DispatchProps & OwnProps

export default function UidTimeline({logs, log, dispatch}: Props) {
  if (logs.length === 0) return null

  const xScale = createScale(logs)

  function queryForAll() {
    dispatch(SearchBar.clearSearchBar())
    dispatch(SearchBar.changeSearchBarInput(log.correlationId()))
    dispatch(submitSearch())
  }

  return (
    <div className="uid-waterfall">
      {logs.map((currLog, i) => (
        <PathRow
          key={i}
          log={currLog}
          current={isEqual(currLog, log)}
          position={xScale(currLog.cast("ts") as Date)}
          onClick={() => dispatch(viewLogDetail(currLog))}
        />
      ))}
      <div className="caption">
        <p className="data-label">{captionText(logs, queryForAll)}</p>
      </div>
    </div>
  )
}

function createScale(logs) {
  const tss = []

  for (const log of logs) {
    tss.push(log.cast("ts"))
  }

  let [start, end] = d3.extent(tss)
  if (start === end)
    end = brim
      .time(start)
      .add(1, "ms")
      .toDate()

  return d3
    .scaleTime()
    .domain([start, end])
    .range([0, 100])
}

function PathRow({log, current, position, ...rest}) {
  const ts = log.cast("ts")
  const path = log.getString("_path")
  return (
    <div className="waterfall-row" {...rest}>
      <div className="data-label">{brim.time(ts).format("HH:mm:ss.SSS")}</div>
      <div className="slider">
        <div className="line" />
        <span
          className={classNames("path-tag", `${path}-bg-color`, {current})}
          style={{left: position + "%"}}
        >
          {path}
        </span>
      </div>
    </div>
  )
}

function captionText(logs: Log[], queryForAll) {
  const limit = logs.length === UID_CORRELATION_LIMIT
  const conn = logs.find((l) => l.getString("_path") === "conn")

  if (limit)
    return (
      <>
        Limited to 100 events. <a onClick={queryForAll}>Query for all.</a>
      </>
    )
  else if (conn) {
    return `Duration: ${conn.cast("duration") || 0}s`
  } else {
    return "Duration: 0s (No conn log)"
  }
}

export const XUidTimeline = connect(null, dispatchToProps)(UidTimeline)
