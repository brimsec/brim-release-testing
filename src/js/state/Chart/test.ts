import Tabs from "../Tabs"
import chart from "./"
import initTestStore from "../../test/initTestStore"

let store, tabId
beforeEach(() => {
  store = initTestStore()
  tabId = Tabs.getActive(store.getState())
})

const records = [
  [
    {name: "ts", type: "string", value: "0"},
    {name: "_path", type: "string", value: "conn"},
    {name: "count", type: "string", value: "500"}
  ],
  [
    {name: "ts", type: "string", value: "100"},
    {name: "_path", type: "string", value: "dns"},
    {name: "count", type: "string", value: "300"}
  ]
]

test("chart records append", () => {
  const state = store.dispatchAll([
    chart.appendRecords(tabId, [records[0]]),
    chart.appendRecords(tabId, [records[1]])
  ])

  expect(chart.getData(state)).toEqual({
    keys: ["conn", "dns"],
    table: {"0": {conn: 500}, "100000": {dns: 300}}
  })
})

test("chart records remains unique", () => {
  const state = store.dispatchAll([
    chart.appendRecords(tabId, [records[0]]),
    chart.appendRecords(tabId, [records[0]])
  ])

  expect(chart.getData(state)).toEqual({
    keys: ["conn"],
    table: {"0": {conn: 500}}
  })
})

test("chart records status", () => {
  const state = store.dispatchAll([chart.setStatus(tabId, "SUCCESS")])

  expect(chart.getStatus(state)).toBe("SUCCESS")
})

test("chart records clear", () => {
  const state = store.dispatchAll([
    chart.appendRecords(tabId, records),
    chart.setStatus(tabId, "SUCCESS"),
    chart.clear(tabId)
  ])

  expect(chart.getStatus(state)).toBe("INIT")
  expect(chart.getData(state)).toEqual({keys: [], table: {}})
})
