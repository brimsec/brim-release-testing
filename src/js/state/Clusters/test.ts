import Clusters from "./"
import initTestStore from "../../test/initTestStore"

let store
beforeEach(() => {
  store = initTestStore()
})

const cluster = {
  id: "123",
  host: "boom.com",
  port: "9867",
  username: "kerr",
  password: "123"
}

test("addCluster", () => {
  const state = store.dispatchAll([Clusters.add(cluster)])

  expect(Clusters.id("123")(state)).toEqual(cluster)
})

test("addCluster when it already exists", () => {
  const state = store.dispatchAll([
    Clusters.add(cluster),
    Clusters.add(cluster)
  ])

  expect(Clusters.all(state)).toEqual([cluster])
})

test("removeCluster", () => {
  const state = store.dispatchAll([
    Clusters.add(cluster),
    Clusters.remove("123")
  ])

  expect(Clusters.all(state)).toEqual([])
})
