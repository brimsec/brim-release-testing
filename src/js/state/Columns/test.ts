import Columns from "./"
import initTestStore from "../../test/initTestStore"

const tableId = "test"
let store
beforeEach(() => {
  store = initTestStore()
})
describe("Columns", () => {
  test("get initial state", () => {
    const state = store.getState()
    expect(Columns.getColumns(state)).toEqual({})
  })

  test("Bulk update column settings", () => {
    const state = store.dispatchAll([
      Columns.updateColumns(tableId, {
        "_path:string": {
          width: 22,
          isVisible: true,
          position: 0
        },
        "ts:time": {
          width: 200,
          isVisible: false,
          position: 1
        }
      }),
      Columns.updateColumns(tableId, {
        "_path:string": {isVisible: true, width: 100}
      })
    ])

    expect(Columns.getColumns(state)[tableId]).toEqual({
      "_path:string": {
        width: 100,
        isVisible: true,
        position: 0
      },
      "ts:time": {
        width: 200,
        isVisible: false,
        position: 1
      }
    })
  })

  test("hide one column", () => {
    const state = store.dispatchAll([
      Columns.hideColumn(tableId, {name: "a", type: "string"})
    ])

    const table = Columns.getColumns(state)[tableId]

    expect(table["a:string"]).toEqual({isVisible: false})
  })

  test("show one column", () => {
    const state = store.dispatchAll([
      Columns.hideColumn(tableId, {name: "a", type: "string"}),
      Columns.showColumn(tableId, {name: "a", type: "string"})
    ])

    const table = Columns.getColumns(state)[tableId]

    expect(table["a:string"]).toEqual({isVisible: true})
  })

  test("show all columns", () => {
    store.dispatch(
      Columns.updateColumns(tableId, {
        "a:string": {isVisible: false},
        "b:string": {isVisible: false},
        "c:string": {isVisible: false}
      })
    )

    const state = store.dispatchAll([Columns.showAllColumns(tableId)])

    const table = Columns.getColumns(state)[tableId]

    expect(table["a:string"]).toEqual({isVisible: true})
    expect(table["b:string"]).toEqual({isVisible: true})
    expect(table["c:string"]).toEqual({isVisible: true})
  })

  test("hide all columns", () => {
    store.dispatch(
      Columns.updateColumns(tableId, {
        "a:string": {isVisible: true},
        "b:string": {isVisible: true},
        "c:string": {isVisible: false}
      })
    )

    const state = store.dispatchAll([Columns.hideAllColumns(tableId)])

    const table = Columns.getColumns(state)[tableId]

    expect(table["a:string"]).toEqual({isVisible: false})
    expect(table["b:string"]).toEqual({isVisible: false})
    expect(table["c:string"]).toEqual({isVisible: false})
  })
})
