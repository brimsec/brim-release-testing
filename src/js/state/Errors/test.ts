import Errors from "./"
import initTestStore from "../../test/initTestStore"

let store
beforeEach(() => {
  store = initTestStore()
})

test("ERROR_CREATE", () => {
  const state = store.dispatchAll([Errors.createError("Bug")])

  expect(Errors.getErrors(state)[0]).toEqual({
    type: "AppError",
    message: "Bug",
    details: []
  })
})

test("ERRORS_CLEAR", () => {
  const state = store.dispatchAll([
    Errors.createError("Bug"),
    Errors.createError("Bug2"),
    Errors.clearErrors()
  ])

  expect(Errors.getErrors(state)).toEqual([])
})
