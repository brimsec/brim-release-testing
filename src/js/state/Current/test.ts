import Clusters from "../Clusters"
import Current from "./"
import Spaces from "../Spaces"
import fixtures from "../../test/fixtures"
import initTestStore from "../../test/initTestStore"

let store

beforeEach(() => {
  store = initTestStore()
})

test("setting the space id", () => {
  store.dispatch(Current.setSpaceId("1"))

  expect(Current.getSpaceId(store.getState())).toBe("1")
})

test("setting the connection id", () => {
  store.dispatch(Current.setConnectionId("a"))

  expect(Current.getConnectionId(store.getState())).toBe("a")
})

test("getting the actual connection", () => {
  const conn = {
    id: "myconn",
    host: "www.myconn.com",
    port: "123",
    username: "",
    password: ""
  }
  const state = store.dispatchAll([
    Clusters.add(conn),
    Current.setConnectionId("myconn")
  ])

  expect(Current.mustGetConnection(state)).toEqual(conn)
})

test("getting the actual space", () => {
  const space = fixtures("space1")
  const conn = {
    id: "myconn",
    host: "www.myconn.com",
    port: "123",
    username: "",
    password: ""
  }
  const state = store.dispatchAll([
    Clusters.add(conn),
    Spaces.setDetail("myconn", space),
    Current.setConnectionId("myconn"),
    Current.setSpaceId(space.id)
  ])

  expect(Current.mustGetSpace(state)).toEqual(expect.objectContaining(space))
})
