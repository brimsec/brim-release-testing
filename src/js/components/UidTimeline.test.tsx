import React from "react"
import {shallow} from "enzyme"
import UidTimeline from "./UidTimeline"
import * as mockLogs from "../test/mockLogs"

test("shapshot", () => {
  const wrapper = shallow(
    <UidTimeline
      log={mockLogs.conn()}
      logs={[mockLogs.conn(), mockLogs.dns(), mockLogs.http()]}
      dispatch={jest.fn()}
    />
  )
  expect(wrapper).toMatchSnapshot()
})
