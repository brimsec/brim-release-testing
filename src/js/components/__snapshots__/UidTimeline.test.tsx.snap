// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shapshot 1`] = `
<div
  className="uid-waterfall"
>
  <PathRow
    current={true}
    key="0"
    log={
      Log {
        "descriptor": Array [
          Object {
            "name": "_td",
            "type": "int",
          },
          Object {
            "name": "_path",
            "type": "string",
          },
          Object {
            "name": "ts",
            "type": "time",
          },
          Object {
            "name": "uid",
            "type": "string",
          },
          Object {
            "name": "id.orig_h",
            "type": "addr",
          },
          Object {
            "name": "id.orig_p",
            "type": "port",
          },
          Object {
            "name": "id.resp_h",
            "type": "addr",
          },
          Object {
            "name": "id.resp_p",
            "type": "port",
          },
          Object {
            "name": "proto",
            "type": "enum",
          },
          Object {
            "name": "service",
            "type": "string",
          },
          Object {
            "name": "duration",
            "type": "interval",
          },
          Object {
            "name": "orig_bytes",
            "type": "count",
          },
          Object {
            "name": "resp_bytes",
            "type": "count",
          },
          Object {
            "name": "conn_state",
            "type": "string",
          },
          Object {
            "name": "local_orig",
            "type": "bool",
          },
          Object {
            "name": "local_resp",
            "type": "bool",
          },
          Object {
            "name": "missed_bytes",
            "type": "count",
          },
          Object {
            "name": "history",
            "type": "string",
          },
          Object {
            "name": "orig_pkts",
            "type": "count",
          },
          Object {
            "name": "orig_ip_bytes",
            "type": "count",
          },
          Object {
            "name": "resp_pkts",
            "type": "count",
          },
          Object {
            "name": "resp_ip_bytes",
            "type": "count",
          },
          Object {
            "name": "tunnel_parents",
            "type": "set[string]",
          },
        ],
        "tuple": Array [
          "0",
          "conn",
          "1425612054.369843",
          "Cynwae4qh1GxM82hQ2",
          "192.168.0.50",
          "1900",
          "239.255.255.250",
          "1900",
          "udp",
          "-",
          "2.000293",
          "282",
          "0",
          "S0",
          "-",
          "-",
          "0",
          "D",
          "3",
          "366",
          "0",
          "0",
          "(empty)",
        ],
      }
    }
    onClick={[Function]}
    position={100}
  />
  <PathRow
    current={false}
    key="1"
    log={
      Log {
        "descriptor": Array [
          Object {
            "name": "_td",
            "type": "int",
          },
          Object {
            "name": "_path",
            "type": "string",
          },
          Object {
            "name": "ts",
            "type": "time",
          },
          Object {
            "name": "uid",
            "type": "string",
          },
          Object {
            "name": "id.orig_h",
            "type": "addr",
          },
          Object {
            "name": "id.orig_p",
            "type": "port",
          },
          Object {
            "name": "id.resp_h",
            "type": "addr",
          },
          Object {
            "name": "id.resp_p",
            "type": "port",
          },
          Object {
            "name": "proto",
            "type": "enum",
          },
          Object {
            "name": "trans_id",
            "type": "count",
          },
          Object {
            "name": "rtt",
            "type": "interval",
          },
          Object {
            "name": "query",
            "type": "string",
          },
          Object {
            "name": "qclass",
            "type": "count",
          },
          Object {
            "name": "qclass_name",
            "type": "string",
          },
          Object {
            "name": "qtype",
            "type": "count",
          },
          Object {
            "name": "qtype_name",
            "type": "string",
          },
          Object {
            "name": "rcode",
            "type": "count",
          },
          Object {
            "name": "rcode_name",
            "type": "string",
          },
          Object {
            "name": "AA",
            "type": "bool",
          },
          Object {
            "name": "TC",
            "type": "bool",
          },
          Object {
            "name": "RD",
            "type": "bool",
          },
          Object {
            "name": "RA",
            "type": "bool",
          },
          Object {
            "name": "Z",
            "type": "count",
          },
          Object {
            "name": "answers",
            "type": "vector[string]",
          },
          Object {
            "name": "TTLs",
            "type": "vector[interval]",
          },
          Object {
            "name": "rejected",
            "type": "bool",
          },
        ],
        "tuple": Array [
          "8",
          "dns",
          "1425565514.419939",
          "Cum4LVba3W3KNG6qa",
          "fe80::eef4:bbff:fe51:89ec",
          "5353",
          "ff02::fb",
          "5353",
          "udp",
          "0",
          "0.119484",
          "_ipp._tcp.local",
          "1",
          "C_INTERNET",
          "12",
          "PTR",
          "0",
          "NOERROR",
          "T",
          "F",
          "F",
          "F",
          "0",
          "_workstation._tcp.local,sniffer [ec:f4:bb:51:89:ec]._workstation._tcp.local",
          "4500.000000,4500.000000",
          "F",
        ],
      }
    }
    onClick={[Function]}
    position={0}
  />
  <PathRow
    current={false}
    key="2"
    log={
      Log {
        "descriptor": Array [
          Object {
            "name": "_td",
            "type": "int",
          },
          Object {
            "name": "_path",
            "type": "string",
          },
          Object {
            "name": "ts",
            "type": "time",
          },
          Object {
            "name": "uid",
            "type": "string",
          },
          Object {
            "name": "id.orig_h",
            "type": "addr",
          },
          Object {
            "name": "id.orig_p",
            "type": "port",
          },
          Object {
            "name": "id.resp_h",
            "type": "addr",
          },
          Object {
            "name": "id.resp_p",
            "type": "port",
          },
          Object {
            "name": "trans_depth",
            "type": "count",
          },
          Object {
            "name": "method",
            "type": "string",
          },
          Object {
            "name": "host",
            "type": "string",
          },
          Object {
            "name": "uri",
            "type": "string",
          },
          Object {
            "name": "referrer",
            "type": "string",
          },
          Object {
            "name": "version",
            "type": "string",
          },
          Object {
            "name": "user_agent",
            "type": "string",
          },
          Object {
            "name": "request_body_len",
            "type": "count",
          },
          Object {
            "name": "response_body_len",
            "type": "count",
          },
          Object {
            "name": "status_code",
            "type": "count",
          },
          Object {
            "name": "status_msg",
            "type": "string",
          },
          Object {
            "name": "info_code",
            "type": "count",
          },
          Object {
            "name": "info_msg",
            "type": "string",
          },
          Object {
            "name": "tags",
            "type": "set[enum]",
          },
          Object {
            "name": "username",
            "type": "string",
          },
          Object {
            "name": "password",
            "type": "string",
          },
          Object {
            "name": "proxied",
            "type": "set[string]",
          },
          Object {
            "name": "orig_fuids",
            "type": "vector[string]",
          },
          Object {
            "name": "orig_filenames",
            "type": "vector[string]",
          },
          Object {
            "name": "orig_mime_types",
            "type": "vector[string]",
          },
          Object {
            "name": "resp_fuids",
            "type": "vector[string]",
          },
          Object {
            "name": "resp_filenames",
            "type": "vector[string]",
          },
          Object {
            "name": "resp_mime_types",
            "type": "vector[string]",
          },
        ],
        "tuple": Array [
          "10",
          "http",
          "1425567042.047800",
          "CBQjjR1ZrY6LmUfzX5",
          "192.168.0.51",
          "60677",
          "91.189.89.240",
          "80",
          "1",
          "GET",
          "start.ubuntu.com",
          "/14.10/Google/?sourceid=hp",
          "-",
          "1.0",
          "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:33.0) Gecko/20100101 Firefox/33.0",
          "0",
          "6005",
          "200",
          "OK",
          "-",
          "-",
          "(empty)",
          "-",
          "-",
          "-",
          "-",
          "-",
          "-",
          "FUPWLQXTNsTNvf33",
          "-",
          "text/html",
        ],
      }
    }
    onClick={[Function]}
    position={3.2824014636887235}
  />
  <div
    className="caption"
  >
    <p
      className="data-label"
    >
      Duration: 2.000293s
    </p>
  </div>
</div>
`;
